### ISSUE TITLE
Bouncy Castle PCL & Xamarin Unified Api

### ISSUE THREAD

**Issue by gautambjain** [onovotny/BouncyCastle-PCL#4]
Hello,

Thanks for your PCL. We have been using it successfully with Xamarin.Android & Xamarin.iOS projects.

However, recently after converting our Xamarin.iOS project to support Unified API, the project doesn't compile due to crypto assembly. This issue has cropped up only after converting the project to Unified API. Earlier Bouncy Castle was working with Xamarin.iOS.

Is this a known issue? Is there a fix/workaround already?

Regards,
Gautam Jain

**Comment by onovotny** [comment]
Hi Guatam,

The BouncyCastle PCL is a single library for all platforms. I'm not aware of anything specific that would affect just the iOS Unified profile. 

What is the compilation error?

**Comment by gautambjain** [comment]
Thanks Oren,

I am getting following error after upgrading to iOS Unified.

Error MT2002: Failed to resolve assembly: 'crypto, Version=1.7.0.0,
Culture=neutral, PublicKeyToken=null' (MT2002)

Regards,
Gautam Jain

**Comment by onovotny** [comment]
Hi Guatam,

This error appears to be from the Xamarin linker -- I would suggest opening a support ticket with Xamarin as it appears to be a Xamarin bug/issue. This particular PCL does not have any platform-specific code in it; it's a "pure" PCL.

Regards

**Comment by bscheiman** [comment]
I believe the .nuspec is missing the xamarinios10 platform (ie, +xamarinios10). I'll try adding the full enchilada and get back w/ a PR if it solves this particular problem.

**Comment by onovotny** [comment]
It should not need that for a pcl as nuget auto assumes compat. Otherwise every nuget that's a pure pcl without platform sporadic specific bits would need to be updated, and that doesn't scale.
________________________________

**Comment by bscheiman** [comment]
True, but it also fixes the issue on an immediate basis. On my end, Xamarin Studio (Xam.iOS 8.8.1.9) requires an assembly for .NET 4.5 + Windows 8. The original nuget only defines .NET 4.0 + Windows 8.

**Comment by onovotny** [comment]
Strange....if you send in a PR, please include MonoTouch and MonoAndroid as well.

**Comment by bscheiman** [comment]
Sure thing. PR coming your way.

**Comment by bscheiman** [comment]
As a side note, replacing net4 with net45 should do the trick as well. Not sure you want to exclude .NET 4.0 though.

**Comment by onovotny** [comment]
Yeah, I didn't see the point of moving it net45 when there's nothing 4.5 specific in it :/

Are you using the 1.7.x stable release or the 1.8 beta? I can release a new set of nupkg's.

**Comment by bscheiman** [comment]
1.7.x atm. That would be great. Thanks!

**Comment by gautambjain** [comment]
>>>> Error MT2002: Failed to resolve assembly: 'crypto, Version=1.7.0.0,
Culture=neutral, PublicKeyToken=null' (MT2002)

I was getting above error when I upgraded my project to Xamarin Unified
API. BouncyCastle PCL as included inside another PCL (this actually
compiled correctly before Unified API).

Then I moved BouncyCastle PCL to Xamarin.Android & Xamarin.iOS projects.
This solved the problem.

Thanks all for your help.

Regards,
Gautam Jain
